### ISSUE TITLE
iconv(): Detected an incomplete multibyte character in input string

### ISSUE THREAD

**Issue by vanBrunneren** [PHPOffice/PhpSpreadsheet#494]
This is:

```
- [x ] a bug report
- [ ] a feature request
- [ ] **not** a usage question (ask them on https://stackoverflow.com/questions/tagged/phpspreadsheet or https://gitter.im/PHPOffice/PhpSpreadsheet)
```

### What is the expected behavior?
no error on FileImport

### What is the current behavior?
We're importing a .csv File with a lot (~1000) entries and sometimes (not reconstructable) we're receiving the following error:

iconv(): Detected an incomplete multibyte character in input string
File: PHPSpreadsheet/vendor/phpoffice/phpspreadsheet/src/PhpSpreadsheet/Shared/StringHelper.php
Line: 346

if we replace the content of this function with the following solution, the error disappears:
```
 public static function sanitizeUTF8($value)
    {
        if (self::getIsIconvEnabled()) {

            // NEW ----------------------------------------------------------------
            $encoding = mb_detect_encoding($value, mb_detect_order(), false);

            if($encoding == "UTF-8")
            {
                $value = mb_convert_encoding($value, 'UTF-8', 'UTF-8');    
            }

            $value = iconv(mb_detect_encoding($value, mb_detect_order(), false), "UTF-8//IGNORE", $value);
            // --------------------------------------------------------------------

            // OLD --------------------------------------
            // $value = @iconv('UTF-8', 'UTF-8', $value);
            // -------------------------------------------
            return $value;
        }

        $value = mb_convert_encoding($value, 'UTF-8', 'UTF-8');

        return $value;
    }
```

Is this a known issue and are there plans to change this in your library? This is not a very important issue because it appears randomly and we have a workaround. But it would be nice if this would get fixed, so we have a fully functional library.

### Which versions of PhpSpreadsheet and PHP are affected?
1.2.1.0

**Comment by dinari01** [comment]
Hey,

What was the file extension of the excel file? 
I was able to avoid this issue by converting the file to a "xlsx" format.

**Comment by loumzy** [comment]
Check if you excel contain image, or other file，

**Comment by oleibman** [comment]
Addressed by PR #2994 (Aug. 2022) and PR #2910 (July 2022).
